code/z_adventurezones/lavamoon.dm

```diff
@@ -883,8 +893,16 @@ var/global/iomoon_blowout_state = 0 //0: Hasn't occurred, 1: Moon is irradiated
 				if (spawners)
 					for (var/obj/iomoon_boss/bot_spawner/aSpawner in spawners)
 						aSpawner.spawn_bot()
-				if (rotors)
-					rotors.icon_state = "powercore_rotors_fast"
+				if (base)
+					base.icon_state = "powercore_base_fast"
+					top.icon_state = "powercore_core_hit"
+					sleep(0.6 SECONDS)
+					top.icon_state = "powercore_core_fast"
+			else
+				if (base || iomoon_blowout_state == 1)
+					top.icon_state = "powercore_core_hit"
+					sleep(0.6 SECONDS)
+					top.icon_state = "powercore_core_active"
 
 
 		attack_hand(var/mob/user)
```

Comment:
Everything below the `playsound` seems unnecessary in attack_hand? We aren't doing damage so I don't see why there is a health check, bot spawning and attempted icon updates.